### PR TITLE
Automatically mark received HUs carrying BOM components as source HUs.

### PR DESCRIPTION
**Pull request by pvpurcarcosmin**
https://github.com/metasfresh/metasfresh/issues/6987

### CODE REVIEW

File: backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/inout/impl/ReceiptInOutLineHUAssignmentListener.java [hunk 1]
```diff
@@ -54,6 +58,7 @@
 public final class ReceiptInOutLineHUAssignmentListener extends HUAssignmentListenerAdapter
 {
 	public static final transient ReceiptInOutLineHUAssignmentListener instance = new ReceiptInOutLineHUAssignmentListener();
+	private static final SourceHUsService sourceHuService = SpringContextHolder.instance.getBean(SourceHUsService.class);
```
Comment (metas-ts) [inline backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/inout/impl/ReceiptInOutLineHUAssignmentListener.java:61]: tbh the "static" makes be a bit nervous..

File: backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/sourcehu/SourceHUsService.java [hunk 2]
```diff
@@ -210,6 +214,25 @@ public boolean isSourceHu(final HuId huId)
 		return Services.get(ISourceHuDAO.class).isSourceHu(huId);
 	}
 
+	public void addSourceHUMarkerIfRequired(@NonNull final HuId huId, @NonNull final ProductId productId, @NonNull final WarehouseId warehouseId)
```
Comment (metas-ts) [inline backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/sourcehu/SourceHUsService.java:217]: I think this should be clearer.
If nothing else, pls add some javadoc here; or improve the method name and e.g. extract `product.isBOM()` into a variable `notAComponent`
Think of a dev wondering why tf it doesn't want to mark a bom-product.

Comment (metas-ts) [reply backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/sourcehu/SourceHUsService.java:217]: PS: also check out the failing test at https://jenkins.metasfresh.com/job/metasfresh/job/gh6987/1/testReport/

File: backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/inout/impl/ReceiptInOutLineHUAssignmentListener.java (hunk 1 above)
Comment (pvpurcarcosmin) [reply backend/de.metas.handlingunits.base/src/main/java/de/metas/handlingunits/inout/impl/ReceiptInOutLineHUAssignmentListener.java:61]: yeah, right.. dropping it

**Pull request merged**